front/lib/dust_api.ts

```diff
@@ -0,0 +1,74 @@
+import { Project } from "@app/types/project";
+
+const { DUST_API: DUST_API_URL } = process.env;
+
+type ErrorResponse = {
+  error: {
+    message: string;
+    code: number;
+  };
+};
+type SuccessResponse<T> = { response: T };
+
+type DustAPIResponse<T> = ErrorResponse | SuccessResponse<T>;
+
+export function isErrorResponse<T>(
+  response: DustAPIResponse<T>
+): response is ErrorResponse {
+  return "error" in response;
+}
+
+type GetDatasetsResponse = {
+  datasets: { [key: string]: { hash: string; created: number }[] };
+};
+
+type GetDatasetResponse = {
+  dataset: {
```

Comment:
Shouldn't we use `DatasetType` (extending it if needed)?

Reply:
As far as I can tell, the types are very different and only have `data` in common. Am I missing something ?

Reply:
Ah sorry you're right. types/dataset.rs is associated with the `front` SQL object `Dataset` and is "hydrated" with the data coming from the DustAPI :+1: 

You're perfectly right.

Maybe we want named types in this file as well. DustAPIDataset DustAPIRun DustAPIDataSource etc... ?

That being said, for now this one is used only once so I'm fine without DRY-ing up these types :+1:

Reply:
> Maybe we want named types in this file as well. DustAPIDataset DustAPIRun DustAPIDataSource etc... ?

That sounds good to me